src/profile-logic/tracks.js

```diff
@@ -744,180 +723,169 @@ export function getLocalTrackName(
+  const isInterestingEvenWithMinimalActivity =
+    _isFirefoxMediaThreadWhichIsUsuallyIdle(thread);
+  const sampleScore = _computeThreadSampleScore(
     profile,
     thread,
-    activeSamplePercentage
+    maxCpuDeltaPerInterval
   );
+  const boostedSampleScore = isInterestingEvenWithMinimalActivity
+    ? sampleScore * AUDIO_THREAD_SAMPLE_SCORE_BOOST_FACTOR
+    : sampleScore;
+  return {
+    isEssentialFirefoxThread,
+    isInterestingEvenWithMinimalActivity,
+    sampleScore,
+    boostedSampleScore,
+  };
 }
 
-// Any thread, except content thread with no RefreshDriverTick, with less than
-// 5% non-idle time will be hidden.
-const PERCENTAGE_ACTIVE_SAMPLES = 0.05;
-
-// Any content thread with no RefreshDriverTick with less than 20% non-idle
-// time will be hidden.
-const PERCENTAGE_ACTIVE_SAMPLES_NON_PAINT = 0.2;
-
-/**
- * This function goes through all of the samples in the thread, and sees if some large
- * percentage of them are idle. If the thread is mostly idle, then it should be hidden.
- */
-function _isThreadMostlyFullOfIdleSamples(
-  profile: Profile,
-  thread: Thread,
-  activeSamplePercentage: number
-): boolean {
-  let maxActiveStackCount = activeSamplePercentage * thread.samples.length;
-  let activeStackCount = 0;
-  let filteredStackCount = 0;
-
-  const { categories } = profile.meta;
-  if (!categories) {
-    // Profiles that are imported may not have categories. In this case do not try
-    // and deduce anything about idleness.
-    return false;
-  }
-
-  for (
-    let sampleIndex = 0;
-    sampleIndex < thread.samples.length;
-    sampleIndex++
-  ) {
-    const stackIndex = thread.samples.stack[sampleIndex];
-    if (stackIndex === null) {
-      // This stack was filtered out. Most likely this will never actually happen
-      // on a new profile, but keep this check here since the stacks are possibly
-      // null in the Flow type definitions.
-      filteredStackCount++;
-      // Adjust the maximum necessary active stacks to find based on null stacks.
-      maxActiveStackCount =
-        activeSamplePercentage * (thread.samples.length - filteredStackCount);
-    } else {
-      const categoryIndex = thread.stackTable.category[stackIndex];
-
-      const category = categories[categoryIndex];
-      if (category.name !== 'Idle') {
-        activeStackCount++;
-        if (activeStackCount > maxActiveStackCount) {
-          return false;
-        }
-      }
-    }
-  }
+function _isEssentialFirefoxThread(thread: Thread): boolean {
+  return (
+    // Don't hide the Renderer thread. This is because Renderer thread is pretty
+    // useful for understanding the painting with WebRender and it's an important
+    // thread for the users.
+    thread.name === 'Renderer' ||
+    // Don't hide the main thread of the parent process.
+    (thread.name === 'GeckoMain' &&
+      thread.processType === 'default' &&
+      (!thread.processName || thread.processName === 'Parent Process')) ||
```

Comment:
Oh looks like we have another check for `Parent Process` name here now compared to the old check. It's a good idea.

Reply:
Well, it's mostly a workaround for `perfrecord` profiles, which use `"name": "GeckoMain", "processType": "default"` on all their main threads. This workaround would become unnecessary once we have a proper `isMainThread` field and don't need to use the special `GeckoMain` name.

Reply:
Yeah, I would like to have the `isMainThread` field as well.